### PR TITLE
Adding X-RELEVANT-ROLES header

### PR DESCRIPTION
**Pull request by dmnjohns**
This PR adds support for relevant roles. Relevant roles are roles which grant a request access to a resource. Technically, these roles are the intersection of roles which belong to the user (i.e., the values of the X-Roles header) and roles which are required to access a resource (i.e., the values of rax:roles). Relevant roles will be added to the X-Relevant-Roles request header.

### CODE REVIEW

**Comment by dmnjohns** [comment]
The failing tests appear to be tests that verify the path through the state machine. My change added another step in the path. What would you like to do with those tests? For the moment, I might just add the state to the assertions to get the tests to pass.

**Comment by RackerWilliams** [comment]
@dmnjohns  : Please fix the tests by adding the correct assertions.

File: RELEASE.md [hunk 1]
```diff
@@ -3,6 +3,9 @@
 1. Fixed a bug where rax:roles were not masked correctly if default headers were set.
 1. Added support for ```rax:captureHeader```, this allows setting XPath 3.1 paths at the method request, representation, resource, and resources
    level to allow capturing a header with the result of the XPath. The XPaths can simultaneously introspect headers, uri, methods, and body content (xml or json).
+1. Added support for `X-Relevant-Roles`, a header which will be added to the request any time `rax:roles` are being used.
+   `X-Relevant-Roles` are the values of the `X-Roles` header which match a `rax:roles` value for the resource being requested.
+   In other words, `X-Relevant-Roles` are the user roles which granted access to the resource.
```
Comment (RackerWilliams) [inline RELEASE.md]: You should mention that capture head should be enabled for relevant roles to work.

**Comment by dmnjohns** [comment]
I believe I have addressed all of the code review issues.

While fixing the tests, I actually noticed that the XSLT code I had written did not order the sequence set used in the XPath expression it generated, which led to more states being created than were necessary. So I am now ordering the sequence set.

As for the `TestConfig`, it appears that `enableCaptureHeaderExtension` defaults to true anyway, so setting it explicitly on the configs in `RaxRolesBehaviors` should not change any behavior, but it does make things more explicit. So that's good.

File: core/src/main/resources/xsl/raxRoles.xsl [hunk 2]
```diff
@@ -113,15 +113,26 @@
...
-                            required="true" repeating="true">
-                    <xsl:attribute name="fixed" select="rax:transformNBSP(.)"/>
-                </wadl:param>
-            </xsl:for-each>
-        </xsl:if>
+        <xsl:choose>
+            <xsl:when test="not('#all' = $roles)">
+                <xsl:variable name="escaped" as="xsd:string*" select="for $r in $roles return replace($r, '&apos;&apos;', '&apos;&apos;&apos;&apos;')"/>
+                <xsl:variable name="quoted" as="xsd:string*" select="for $r in $escaped return concat('&apos;&apos;', rax:transformNBSP($r), '&apos;&apos;')"/>
+                <xsl:variable name="ordered" as="xsd:string*"><xsl:perform-sort select="$quoted"><xsl:sort/></xsl:perform-sort></xsl:variable>
```
Comment (RackerWilliams) [inline core/src/main/resources/xsl/raxRoles.xsl]: Nice touch on putting these in order.

**Pull request merged**